### PR TITLE
Update sign-here.html

### PR DESCRIPTION
**Pull request by esd100**
Made sign-here dynamic

### CODE REVIEW

File: sign-here.html [hunk 1]
```diff
@@ -15,6 +15,17 @@
...
       }
+      
+      #canvas {
+        position: inherit;
+        padding:10px;
+        border: 2px solid #000000;
+        margin: auto;
+        width: auto
+        background-color: #FFFFFF;
+        box-shadow: 0 1px 4px rgba(0, 0, 0, 0.27), 0 0 40px rgba(0, 0, 0, 0.08) inset;
```
Comment (CaptainCodeman) [inline sign-here.html:26]: It appears that these styles are specific to your app and they break the demo.

File: sign-here.html [hunk 2]
```diff
@@ -23,19 +34,13 @@
     Polymer({
 
       is: 'sign-here',
+      
+      behaviors: [
+        Polymer.IronResizableBehavior
```
Comment (CaptainCodeman) [inline sign-here.html:39]: This causes errors as the dependency isn't included.

File: sign-here.html [hunk 3]
```diff
@@ -138,13 +143,19 @@
...
       ],
 
+      listeners: {
+        'iron-resize': '_onIronResize'
+      },
+
       ready: function() {
         var ratio =  Math.max(window.devicePixelRatio || 1, 1);
+        var xW = Math.round(0.6 * window.innerWidth);
+        var yH = Math.round(0.4 * window.innerWidth);
```
Comment (CaptainCodeman) [inline sign-here.html:153]: This code is all duplicated again later.

File: sign-here.html [hunk 4]
```diff
@@ -180,7 +192,6 @@
        */
       clear: function() {
         this.signaturePad.clear();
-        this.encodeImage();
```
Comment (CaptainCodeman) [inline sign-here.html:183]: This leaves the image property out of sync with the signature pad / canvas.

File: sign-here.html [hunk 5]
```diff
@@ -235,7 +246,38 @@
         if (this.signaturePad) {
           this.encodeImage();
         }
-      }
+      },
+      
+      _onIronResize: function() {
+        var ratio =  Math.max(window.devicePixelRatio || 1, 1);
+        var xW = Math.round(0.6 * window.innerWidth);
```
Comment (CaptainCodeman) [inline sign-here.html:253]: These values seem arbitrary / magic and again, likely app-specific.

File: sign-here.html [hunk 6]
```diff
@@ -235,7 +246,38 @@
...
+        var yH = Math.round(0.4 * window.innerWidth);
+
+        this.$.canvas.style.width = xW + 'px';
+        this.$.canvas.style.height = yH + 'px';
+        this.$.canvas.width = xW * ratio;
+        this.$.canvas.height = yH * ratio;
+        this.$.canvas.getContext("2d").scale(ratio, ratio);
+
+        this.signaturePad = new SignaturePad(this.$.canvas, {
```
Comment (CaptainCodeman) [inline sign-here.html:262]: You are completely re-creating the signature pad on each resize event.

File: sign-here.html [hunk 7]
```diff
@@ -235,7 +246,38 @@
...
+          minWidth: this.minWidth,
+          maxWidth: this.maxWidth,
+          backgroundColor: this.backgroundColor,
+          penColor: this.penColor,
+          velocityFilterWeight: this.velocityFilterWeight,
+          onBegin: this._onBegin.bind(this),
+          onEnd: this._onEnd.bind(this)
+        });
+
+        this.signaturePad.clear();
```
Comment (CaptainCodeman) [inline sign-here.html:273]: And resizing or changing orientation will wipe the signature.

**Comment by CaptainCodeman** [comment]
Sorry, too many issues to merge this. See code comments.

File: sign-here.html (hunk 7 above)
Comment (esd100) [reply sign-here.html:273]: It actually doesn't wipe the signature. It works perfectly.

**Comment by esd100** [comment]
Well, I did my best and I'm not a programmer but a physician. It's too bad that you are unwilling to actually make an effort to make it dynamic.

**Comment by CaptainCodeman** [comment]
> Well, I did my best and I'm not a programmer but a physician. 

Irrelevant.

> It's too bad that you are unwilling to actually make an effort to make it dynamic.

That is completely inaccurate and unfair. I took the time to help you with HTML 101 type stuff in chat to get you up and running but you simply declared it "unfit" and demanded it be changed but without being able to articulate exactly why or how.

I suggested you raise a github issue after you had given it more thought but instead you expect me to merge a broken commit and then again start being insulting when you don't immediately get your way.

Pathetic.

**Comment by esd100** [comment]
I won't stoop down to your level. Your program sucks the way it is.

File: sign-here.html (hunk 7 above)
Comment (CaptainCodeman) [reply sign-here.html:273]: That's possibly because the missing dependency prevents the event from being reached.

Comment (esd100) [inline sign-here.html]: I honestly take your suggestions with a grain of salt, since you, a programmer by trade, can't make a decent dynamic app.

**Pull request closed without merging**